src/Audio/Effects/IntoxicatedEffect.cs

```diff
@@ -0,0 +1,26 @@
+﻿using OpenTK.Audio.OpenAL;
+
+namespace RPVoiceChat.Audio.Effects
+{
+    public class IntoxicatedEffect : SoundEffect
+    {
+        protected float toxicRate;
+
+        public IntoxicatedEffect(int source) : base(source) {}
+
+        protected override int GenerateEffect()
+        {
+            // WARNING : changing pitch also changes audio playback speed, causing overflow of circular audio buffer.
+            // More advanced implementation is required before using this effect
+            float pitch = toxicRate <= 0.2 ? 1 : 1 - (toxicRate / 5);
+            OALW.Source(source, ALSourcef.Pitch, pitch);
+
+            return (int)EffectType.PitchShifter;
```

Comment:
`SoundEffect.effect` is an internal ID of the OpenAL effect, you get it when you generate new OpenAl effect with `ALC.EFX.GenEffect()`
IntoxicatedEffect doesn't use any effects as of now so we can just throw `NotImplementedException`